components/brave_federated_learning/brave_federated_learning_service.h

```diff
@@ -0,0 +1,49 @@
+/* Copyright 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#ifndef BRAVE_COMPONENTS_BRAVE_FEDERATED_LEARNING_BRAVE_FEDERATED_LEARNING_SERVICE_H_
+#define BRAVE_COMPONENTS_BRAVE_FEDERATED_LEARNING_BRAVE_FEDERATED_LEARNING_SERVICE_H_
+
+#include <memory>
+
+#include "base/timer/timer.h"
+#include "components/prefs/pref_service.h"
+#include "url/gurl.h"
+
+class PrefRegistrySimple;
+
+namespace network {
+class SharedURLLoaderFactory;
+class SimpleURLLoader;
+}  // namespace network
+
+namespace brave {
+
+class BraveOperationalProfiling;
+
+class BraveFederatedLearningService {
+ public:
+  explicit BraveFederatedLearningService(
```

Comment:
not explicit, also needs deleted copy ctor / assignment operator. Also I would add `final` to the class declaration (and to  BraveOperationalProfiling as well)

Reply:
👍